### PR TITLE
feat(blog): add visible RSS feed

### LINKED ISSUE #5477: Add visible link to RSS Feed
Evaluate if the site could use a visible RSS / Feed link.

I didn't see it on the current site or anywhere on nodejs.dev

It could aide quick debugging efforts too as its an important source of information. We could also research if the vercel preview could generate a preview link handy in each PR

@ovflowd told me that the information is currently only in the head 

image

---

_Originally posted by @bmuenzenmeyer in https://github.com/nodejs/nodejs.org/pull/5475#pullrequestreview-1516302074_

### PR DESCRIPTION
**Pull request by AugustinMauroy**
## Description

Adding icon + link to category page of blog

## Validation

it's should be same as [figma design](https://www.figma.com/file/pu1vZPqNIM7BePd6W8APA5/Node.js?type=design&node-id=332-46117&mode=design&t=uJ6QNBLDMnOsHFiu-0)

## Related Issues

Related to #5477

### Check List

- [X] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [X] I have run `npx turbo lint` to ensure the code follows the style guide. And run `npx turbo lint:fix` to fix the style errors if necessary.
- [X] I have run `npx turbo format` to ensure the code follows the style guide.
- [X] I have run `npx turbo test` to check if all tests are passing.
- NA I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

**Comment by bmuenzenmeyer** [comment]
@AugustinMauroy thanks for working on this. while I agree it matches what the figma shows

- ✅ en/feed/blog.xml (everything)
- ✅ en/feed/releases.xml - such as https://nodejs-org-git-featrss-visible-link-openjs.vercel.app/en/feed/releases.xml
- ✅ en/feed/vulnerability.xml - such as https://nodejs-org-git-featrss-visible-link-openjs.vercel.app/en/feed/vulnerability.xml
- ~en/feed/announcements.xml - not sure why this isn't working~

File: layouts/New/Blog.tsx [hunk 1]
```diff
@@ -34,16 +36,28 @@ const BlogLayout: FC = async () => {
     }));
 
   const blogData = await getBlogCategory(pathname);
+  // haky way to determine the rss feed
+  const rssFeed = () => {
+    if (blogData.category === 'vulnerability') return 'vulnerability';
+    if (blogData.category === 'release') return 'releases';
+    return 'blog';
+  };
```
Comment (bmuenzenmeyer) [inline layouts/New/Blog.tsx]: ah, I commented before I saw this code. yes, we should improve upon this. not sure why `announcements` doesnt work but I assume it comes from https://github.com/nodejs/nodejs.org/blob/56799569c7950ab0fe8f69c56364719a4aabfc47/next-data/generators/websiteFeeds.mjs

Comment (AugustinMauroy) [reply layouts/New/Blog.tsx]: I didn't get why do you meant by "doesn't work" because when I do `curl https://nodejs-org-git-featrss-visible-link-openjs.vercel.app/en/feed/blog.xml` I have the feed. and when I click on link I also have correct link.

But I allow this code snipette isn't sexy but idk how to do better

Comment (AugustinMauroy) [reply layouts/New/Blog.tsx]: i had update code to use site config.

File: layouts/New/Blog.tsx [hunk 2]
```diff
@@ -1,11 +1,14 @@
+import { RssIcon } from '@heroicons/react/24/solid';
 import { getTranslations } from 'next-intl/server';
 import type { FC } from 'react';
 
 import { getClientContext } from '@/client-context';
 import WithBlogCategories from '@/components/withBlogCategories';
 import WithFooter from '@/components/withFooter';
 import WithNavBar from '@/components/withNavBar';
+import { Link } from '@/navigation.mjs';
```
Comment (ovflowd) [inline layouts/New/Blog.tsx]: We should use the Link component from components folder

Comment (AugustinMauroy) [reply layouts/New/Blog.tsx]: you mean ?

```suggestion
import Link from '@/components/Link';
```

Comment (ovflowd) [reply layouts/New/Blog.tsx]: https://github.com/nodejs/nodejs.org/blob/main/components/Link.tsx

File: layouts/New/Blog.tsx (hunk 1 above)
Comment (ovflowd) [reply layouts/New/Blog.tsx]: > ah, I commented before I saw this code. yes, we should improve upon this. not sure why announcements doesnt work but I assume it comes from [5679956/next-data/generators/websiteFeeds.mjs](https://github.com/nodejs/nodejs.org/blob/56799569c7950ab0fe8f69c56364719a4aabfc47/next-data/generators/websiteFeeds.mjs)

There's no announcements RSS feed on site config

File: layouts/New/Blog.tsx [hunk 3]
```diff
@@ -41,9 +44,17 @@ const BlogLayout: FC = async () => {
...
       <div className={styles.blogLayout}>
         <main>
-          <h1>{t('layouts.blog.title')}</h1>
-
-          <p>{t('layouts.blog.subtitle')}</p>
+          <header>
+            <h1>
+              {t('layouts.blog.title')}
+              <Link
+                href={`/feed/${siteConfig.rssFeeds.find(item => item.blogCategory === blogData.category)?.file ?? 'blog.xml'}`}
```
Comment (ovflowd) [inline layouts/New/Blog.tsx]: This logic seems broken.

Comment (AugustinMauroy) [reply layouts/New/Blog.tsx]: why do you mean ? I have manually test on local and it's work.

Comment (ovflowd) [reply layouts/New/Blog.tsx]: It doesn't work on the preview branch. The preview branch is the closest to a production environment.

Comment (ovflowd) [reply layouts/New/Blog.tsx]: "I have manually test on local and it's work." 

https://www.reddit.com/r/ProgrammerHumor/comments/70we66/it_works_on_my_machine/

Comment (AugustinMauroy) [reply layouts/New/Blog.tsx]: FOUND it's because type annotation isn't right.

Comment (ovflowd) [reply layouts/New/Blog.tsx]: Looks broken on this category https://nodejs-org-git-featrss-visible-link-openjs.vercel.app/en/blog/announcements, shouldn't it show `blog.xml`?

Comment (AugustinMauroy) [reply layouts/New/Blog.tsx]: > shouldn't it show blog.xml?

yeah, I think

File: layouts/New/Blog.tsx [hunk 4]
```diff
@@ -41,9 +44,17 @@ const BlogLayout: FC = async () => {
...
       <div className={styles.blogLayout}>
         <main>
-          <h1>{t('layouts.blog.title')}</h1>
-
-          <p>{t('layouts.blog.subtitle')}</p>
+          <header>
+            <h1>
+              {t('layouts.blog.title')}
+              <Link
+                href={`/feed/${siteConfig.rssFeeds.find(item => item.category === blogData.category)?.file ?? 'blog.xml'}`}
```
Comment (ovflowd) [inline layouts/New/Blog.tsx]: Can you please make this a variable? Super weird to have such a long line here

File: layouts/New/Blog.tsx
Comment (ovflowd) [inline layouts/New/Blog.tsx:1]: Please ensure this looks good on mobile and different screen sizes

File: layouts/New/Blog.tsx [hunk 5]
```diff
@@ -41,9 +44,17 @@ const BlogLayout: FC = async () => {
 
       <div className={styles.blogLayout}>
         <main>
-          <h1>{t('layouts.blog.title')}</h1>
-
-          <p>{t('layouts.blog.subtitle')}</p>
+          <header>
```
Comment (ovflowd) [inline layouts/New/Blog.tsx]: This header should probably be extrated into its own component, and then unit tested

File: layouts/New/Blog.tsx
Comment (AugustinMauroy) [reply layouts/New/Blog.tsx:1]: On Figma, the icon is only visible on the desktop light design. So I had to "imporvise". According to my tests, it looks good for mobile.
So if you have any input on design tell me.
BTW: for the hover effect, I used the one on the nav bar buttons.

File: layouts/New/Blog.tsx (hunk 5 above)
Comment (AugustinMauroy) [reply layouts/New/Blog.tsx]: let's do that 🚀

File: layouts/New/Blog.tsx
Comment (ovflowd) [reply layouts/New/Blog.tsx:1]: Can yu attach screenshot for mobile?

File: components/Common/BlogHeader/index.tsx
Comment (ovflowd) [inline components/Common/BlogHeader/index.tsx:1]: This is not a "common" component. Should be under `components/Blog/` instead of `components/Common`

Comment (AugustinMauroy) [reply components/Common/BlogHeader/index.tsx:1]: OK, but I have a question, why put `blogCard` under `common` ?

Comment (ovflowd) [reply components/Common/BlogHeader/index.tsx:1]: Wasn't me who made that component. I think the name of that component is misleading. (https://github.com/nodejs/nodejs.org/blob/main/components/Common/BlogPostCard/index.tsx) that component should probably be just named "ArticleCard" or "PostCard". Because it's not just for Blog, can be reused in other places, such as on the Events page (which I did)

File: layouts/New/Blog.tsx
Comment (AugustinMauroy) [reply layouts/New/Blog.tsx:1]: PR description has been updated

**Comment by ovflowd** [comment]
FYI lint/tests failing @AugustinMauroy

**Comment by ovflowd** [comment]
FYI I've fixed on `main` the blog category layout not respecting mobile width's

File: layouts/New/layouts.module.css [hunk 6]
```diff
@@ -141,27 +141,17 @@
 .blogLayout {
   @apply flex
     w-full
+    max-w-8xl
```
Comment (ovflowd) [inline layouts/New/layouts.module.css]: Why did you add this? This broken the auto margins 🤔

Comment (AugustinMauroy) [reply layouts/New/layouts.module.css]: it's from [this line](https://github.com/nodejs/nodejs.org/pull/6350/files/88e9845728b2ccd013c9eff43fdc90534c48436a#diff-c4b4ff7c439381cc3dfccce4a24420a9b67403bbabb4e5302066fd07f6e272c1L151) is when I merged main and `blogLayout`.

Comment (ovflowd) [reply layouts/New/layouts.module.css]: But... why did you merged them? I understand from which line it is, but could you please undo this change? (the merge?)

There's a reason why there were two different classes.

Comment (AugustinMauroy) [reply layouts/New/layouts.module.css]: I had merged 'cause on JSX we had something like that 
```tsx
<div className={styles.blogLayout}>
  <main>
    {...main}
  </main>
</div>
```

Comment (AugustinMauroy) [reply layouts/New/layouts.module.css]: > This broken the auto margins 🤔

I came back home but I don't know what you mean by "This broken the auto margins".

Capture d’écran 2024-02-19 à 17 07 45

Comment (ovflowd) [reply layouts/New/layouts.module.css]: > I came back home but I don't know what you mean by "This broken the auto margins".

Check on Desktop ;) we align this container always on the center. Because we use "auto margin" alignment (`margin: 0 auto` or `align-items: center` when using flex)

The moment that you impose a fixed width, the auto-centering breaks.

File: layouts/New/layouts.module.css [hunk 7]
```diff
@@ -142,26 +142,15 @@
...
     bg-gradient-subtle
+    px-4
+    py-12
     dark:bg-gradient-subtle-dark
+    md:px-14
+    lg:px-28
     xs:bg-none
     xs:dark:bg-none;
-
-  main {
```
Comment (ovflowd) [inline layouts/New/layouts.module.css]: As I asked before, and I'll ask again, please revert these unrelated changes.

Comment (ovflowd) [reply layouts/New/layouts.module.css]: There are a reason why this ws done this way, and if there's no real reason behind this change, undo it. And make another PR with this change were we can discuss about it.

File: layouts/New/Blog.tsx [hunk 8]
```diff
@@ -38,13 +39,9 @@ const BlogLayout: FC = async () => {
   return (
     <>
       <WithNavBar />
-
```
Comment (ovflowd) [inline layouts/New/Blog.tsx:41]: Why removing these extra spaces? They are intentional to make it easier to read the file / separation of stuff

**Review by ovflowd — APPROVED** [review]
Besides my last comment, LGTM! ✨

File: layouts/New/Blog.tsx (hunk 8 above)
Comment (AugustinMauroy) [reply layouts/New/Blog.tsx:41]: > They are intentional to make it easier to read the file / separation of stuff

😁 it's funny but I'd rather not have any extra space because at least you have everything together. And it makes it easier to see the JSX structure (IMHO).

So just keep extra spaces or not ?

Comment (ovflowd) [reply layouts/New/Blog.tsx:41]: As usual, let's avoid unrelated changes even more ones that are down to personal opinions. We can always open a discussion about what would be better to be adopted.

Comment (AugustinMauroy) [reply layouts/New/Blog.tsx:41]: OK !

File: layouts/New/layouts.module.css [hunk 9]
```diff
@@ -154,13 +154,6 @@
       py-12
       md:px-14
       lg:px-28;
-
-    p {
```
Comment (ovflowd) [inline layouts/New/layouts.module.css:158]: Any reason why this was removed?
